### PR TITLE
TMT6plex mod for MSGF+

### PR DESCRIPTION
**Pull request by jgriss**
These changes were necessary for me to get a TMT10plex experiment working where I used MGSF+.
Additionally, there were some minor changes necessary to get (most of) the reports generated.

### CODE REVIEW

File: R/report-utils-xls.R [hunk 1]
```diff
@@ -439,7 +440,15 @@ write.xls.report <- function(properties.env,report.env,file="isobar-analysis.xls
       x <- x[x[,'end.pos'] <= protein.length,]
       if (nrow(x) == 0) return(c(seq.cov=NA))
       seqq <- rep(FALSE,protein.length);
-      for (i in seq_along(nrow(x))) seqq[x[i,'start.pos']:x[i,'end.pos']] <-  TRUE
+      for (i in seq_along(nrow(x))) {
```
Comment (alyst) [inline R/report-utils-xls.R]: Both for performance and correctness it's better to remove those peptides from `x` before entering the loop -- at the same line, where `x[,'end.pos'] <= protein.length` is being checked (actually, `end.pos` should be checked for NA before that one check). If there are no valid peptides it's better to have `seq.cov=NA` than `seq.cov=0`.

**Comment by alyst** [comment]
Github reports there are merge conflicts, could you please rebase your branch onto the recent master?

File: R/isobar-import.R [hunk 2]
```diff
@@ -1443,6 +1449,7 @@ read.mzid <- function(filename) {
 .convert.msgfp.pepmodif <- function(peptide,modif.masses=
                                       c(iTRAQ4plex=144.102,Cys_CAM=57.021,Oxidation=15.995,
                                         iTRAQ4_ACET="144.102+42.011",ACET=42.011,
+					TMT6plex=229.163,
```
Comment (alyst) [inline R/isobar-import.R]: Could you please adjust indentation?

File: R/isobar-import.R [hunk 3]
```diff
@@ -1422,8 +1422,14 @@ read.mzid <- function(filename) {
     ib.data <- filename
   else
     id.data <- read.delim(filename, sep="\t", stringsAsFactors=FALSE)
+
+  if (! "PepQValue" %in% colnames(id.data)) {
+    stop("Error: No q-values found in result files.\nPlease repeat the MSGF+ search using a\ntarget-decoy search (option -tda 1)")
```
Comment (alyst) [inline R/isobar-import.R]: I think "Error: " should be omitted to match the other isobar error messages.
Also "result files" is not very specific; maybe something like "MSGF+ files"?

**Comment by jgriss** [comment]
@alyst Thanks for the comments! Should be fixed now.

**Comment by fbreitwieser** [comment]
Thank you very much for the pull request and code review, @jgriss and @alyst! If you are fine with it, @alyst, I will merge the pull request.

**Comment by alyst** [comment]
@jgriss Thanks for addressing my comments!
@fbreitwieser I was just tempting to ask to squash the commits, but otherwise it LGTM.

**Comment by fbreitwieser** [comment]
Thank you, merged it!

**Pull request merged**